### PR TITLE
Users/chshrikh/machine group

### PR DESCRIPTION
**Pull request by chshrikh**
Adding capability to configure the agent against machine group

### CODE REVIEW

File: src/Agent.Listener/Configuration/ConfigurationManager.cs [hunk 1]
```diff
@@ -301,8 +312,12 @@ public AgentSettings LoadSettings()
                 PoolName = poolName,
                 ServerUrl = serverUrl,
                 WorkFolder = workFolder,
+                MachineGroup = command.MachineGroup
```
Comment (TingluoHuang) [inline src/Agent.Listener/Configuration/ConfigurationManager.cs]: this should be move into agentProvider.UpdateAgentSetting(settings);

File: src/Agent.Listener/Configuration/ConfigurationManager.cs [hunk 2]
```diff
@@ -381,14 +396,23 @@ public AgentSettings LoadSettings()
...
 
+                    Trace.Info("Agent configured for machineGroup : {0}", settings.MachineGroup.ToString());
+
+                    string agentType = settings.MachineGroup
+                   ? Constants.Agent.AgentConfigurationProvider.DeploymentAgentConfiguration
+                   : Constants.Agent.AgentConfigurationProvider.BuildReleasesAgentConfiguration;
+
+                    var extensionManager = HostContext.GetService<IExtensionManager>();
+                    IConfigurationProvider agentProvider = (extensionManager.GetExtensions<IConfigurationProvider>()).FirstOrDefault(x => x.ConfigurationProviderType == agentType);
+
```
Comment (TingluoHuang) [inline src/Agent.Listener/Configuration/ConfigurationManager.cs]: add ArgUtil.NotNull(agentProvider, agentType);

**Comment by TingluoHuang** [comment]
Looks good, left 2 small comments, make sure you didn't break build/releases. scorch into 1 commit, then merge.

**Comment by TingluoHuang** [comment]
one more thing you might want to think about is what will happen when this agent configure against 2015 tfs, what error message customer will get.

File: src/Agent.Listener/Configuration/ConfigurationManager.cs (hunk 1 above)
Comment (chshrikh) [inline src/Agent.Listener/Configuration/ConfigurationManager.cs]: yes, this is not required here, removing it

File: src/Agent.Listener/Configuration/ConfigurationManager.cs (hunk 2 above)
Comment (chshrikh) [inline src/Agent.Listener/Configuration/ConfigurationManager.cs]: sure

**Pull request merged**